### PR TITLE
feat(cli)!: use clap to read env vars

### LINKED ISSUE #161: CLI: Allow to specify the stronghold file path in the CLI
## Description

title is descriptive enough I think

## Motivation

Similar to the wallet db I think it should be configurable to set this in the CLI. If I am not mistaken then I think currently this is fixed to exist at `./stardust-cli-wallet-db`. It should also be configurable as an environment variable, so that the CLI remains simple to use.

this is what I am imagining:
```
Options:
      --wallet-db-path <PATH>   Set the path to the wallet database [env: WALLET_DATABASE_PATH=] [default: ./stardust-cli-wallet-db]
      --stronghold-path <PATH>  Set the path to stronghold file [env: STRONGHOLD_PATH=] [default: ./stardust-cli-wallet.stronghold]
  -h, --help                    Print help
  -V, --version                 Print version
```

## Open questions (optional)

The environment variable might be debatable, although there is already one for the wallet db, so that would just be consistent I think.

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by Alex6323**
# Description of change

This PR adds CLI options and environments variables to set the wallet db file path, the Stronghold snapshot file path, and the node url.

## Links to any relevant issues

Closes #161 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: cli/src/command/wallet.rs [hunk 1]
```diff
@@ -59,9 +71,9 @@ pub struct InitParameters {
     /// Mnemonic, randomly generated if not provided.
     #[arg(short, long)]
     pub mnemonic: Option<String>,
-    /// Node URL, "https://api.testnet.shimmer.network" if not provided.
-    #[arg(short, long)]
-    pub node: Option<String>,
+    /// Set the node to connect to with this wallet.
+    #[arg(short, long, value_name = "URL", env = "NODE_URL", default_value = DEFAULT_NODE_URL)]
+    pub node_url: String,
```
Comment (thibault-martinez) [inline cli/src/command/wallet.rs:77]: Change the documentation accordingly

File: cli/src/command/wallet.rs [hunk 2]
```diff
@@ -12,14 +12,26 @@ use log::LevelFilter;
...
+    #[arg(long, value_name = "PATH", env = "WALLET_DATABASE_PATH", default_value = DEFAULT_WALLET_DATABASE_PATH)]
+    pub wallet_db_path: String,
+    /// Set the path to the stronghold snapshot file.
+    #[arg(long, value_name = "PATH", env = "STRONGHOLD_SNAPSHOT_PATH", default_value = DEFAULT_STRONGHOLD_SNAPSHOT_PATH)]
+    pub stronghold_snapshot_path: String,
+    /// Set the account to enter.
     pub account: Option<String>,
+    /// Set the log level.
     #[arg(short, long)]
     pub log_level: Option<LevelFilter>,
```
Comment (thibault-martinez) [inline cli/src/command/wallet.rs]: We probably need to have somewhere in the docs where we present these parameters

File: documentation/cli/docs/02_wallet.md [hunk 3]
```diff
@@ -84,7 +84,7 @@ When just initialised, the wallet has no account yet, use the `new` command to c
...
 | ----------- | ----------- |------------------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `mnemonic`  | ✓           | Randomly generated                  | "aunt middle impose faith ramp kid olive good practice motor grab ready group episode oven matrix silver rhythm avocado assume humble tiger shiver hurt" (DO NOT USE THIS MNEMONIC) |
-| `node`      | ✓           | https://api.testnet.shimmer.network | http://localhost:14265                                                                                                                                                              |
+| `node_url`      | ✓           | https://api.testnet.shimmer.network | http://localhost:14265                                                                                                                                                              |
```
Comment (thibault-martinez) [inline documentation/cli/docs/02_wallet.md]: Align!

Comment (Alex6323) [reply documentation/cli/docs/02_wallet.md]: on it

File: cli/src/command/wallet.rs [hunk 4]
```diff
@@ -12,14 +12,26 @@ use log::LevelFilter;
 
 use crate::{error::Error, helper::get_password, println_log_info};
 
+const DEFAULT_NODE_URL: &str = "https://api.testnet.shimmer.network";
+const DEFAULT_WALLET_DATABASE_PATH: &str = "./stardust-cli-wallet-db";
+const DEFAULT_STRONGHOLD_SNAPSHOT_PATH: &str = "./stardust-cli-wallet.stronghold";
```
Comment (Thoralf-M) [inline cli/src/command/wallet.rs:17]: This is a breaking change, do we really want that?

Comment (Alex6323) [reply cli/src/command/wallet.rs:17]: I don't see it. :smile:

Comment (Alex6323) [reply cli/src/command/wallet.rs:17]: Those strings are exactly the same defaults. This PR is breaking due to something else though.

File: documentation/cli/docs/02_wallet.md [hunk 5]
```diff
@@ -79,13 +79,13 @@ The wallet can only be initialised once.
...
-| `node`      | ✓           | https://api.testnet.shimmer.network | http://localhost:14265                                                                                                                                                              |
-| `coin-type` | ✓           | 4219 (=Shimmer)                     | 4218 (=IOTA)                                                                                                                                                                        |
+| Name                 | Description    | Default    | Environment Variable | Example |
+| -------------------- | -------------- |----------- |----------------------| ------- |
+| `--mnemonic` (`-m`)  | Initialises a wallet from an existing mnemonic | ✘ | ✘ |"aunt middle impose ..." |
+| `--node_url` (`-n`)  | Sets the node to connect to | https://api.testnet.shimmer.network | `NODE_URL` | http://localhost:14265 |
+| `--coin-type` (`-c`) | Sets the coin type associated with the wallet  | 4219 (=Shimmer) | ✘ | 4218(=IOTA) |
```
Comment (Thoralf-M) [inline documentation/cli/docs/02_wallet.md]: Should this be node-url or coin_type?

Comment (Alex6323) [reply documentation/cli/docs/02_wallet.md]: oh, nice catch...should be dashes in all cases.

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/examples/client/01_generate_addresses.rs [hunk 6]
```diff
@@ -3,7 +3,7 @@
 
 //! TODO: Example description
 //!
-//! `cargo run --example generate_addresses --release -- [NODE URL]`
+//! `cargo run --example generate_addresses --release -- <NODE_URL>`
```
Comment (thibault-martinez) [inline sdk/examples/client/01_generate_addresses.rs]: Why? Square brackets are for optional parameters and angle brackets are for mandatory parameters. This is optional.

File: sdk/examples/client/node_api_indexer/00_get_basic_outputs.rs [hunk 7]
```diff
@@ -4,7 +4,7 @@
 //! TODO: <insert example description> by calling
 //! `GET api/indexer/v1/outputs/basic`.
 //!
-//! `cargo run --example node_api_indexer_get_basic_outputs --release -- [NODE URL] [ADDRESS]`.
+//! `cargo run --example node_api_indexer_get_basic_outputs --release -- <NODE_URL> <ADDRESS>`.
```
Comment (thibault-martinez) [inline sdk/examples/client/node_api_indexer/00_get_basic_outputs.rs]: Same

File: documentation/cli/docs/02_wallet.md [hunk 8]
```diff
@@ -5,7 +5,17 @@ execution.
...
-## Commands
+You can run `./wallet --help` to print all available CLI commands, arguments, and options. The available options are listed in the table below:
+
+| Name                         | Description | Default | Environment Variable | Example |
+| ---------------------------- | ----------- | -------------------- | --- | --- |
+| `--wallet-db-path`           | Sets the path to the wallet database | `./stardust-cli-wallet-db` | `WALLET_DATABASE_PATH` | `/path/to/wallet-db` |
+| `--stronghold-snapshot-path` | Sets the path to the Stronghold snapshot file | `./stardust-cli-wallet.stronghold` | `STRONGHOLD_SNAPSHOT_PATH`| `/path/to/stronghold-snapshot` |
+| `--log-level` (`-l`)         | Sets the log verbosity level (error, warn, info, debug, trace) | `debug` | ✘ | `trace` |
+| `--help` (`-h`)              | Shows all available commands, arguments and options  | ✘ | ✘ | ✘ |
+| `--version` (`-V`)           | Shows the current version | ✘ | ✘ | ✘ |
```
Comment (thibault-martinez) [inline documentation/cli/docs/02_wallet.md]: Can you align the whole table like the others pls

File: documentation/cli/docs/02_wallet.md [hunk 9]
```diff
@@ -79,13 +79,13 @@ The wallet can only be initialised once.
...
-| `node`      | ✓           | https://api.testnet.shimmer.network | http://localhost:14265                                                                                                                                                              |
-| `coin-type` | ✓           | 4219 (=Shimmer)                     | 4218 (=IOTA)                                                                                                                                                                        |
+| Name                 | Description    | Default    | Environment Variable | Example |
+| -------------------- | -------------- |----------- |----------------------| ------- |
+| `--mnemonic` (`-m`)  | Initialises a wallet from an existing mnemonic | ✘ | ✘ |"aunt middle impose ..." |
+| `--node-url` (`-n`)  | Sets the node to connect to | https://api.testnet.shimmer.network | `NODE_URL` | http://localhost:14265 |
+| `--coin-type` (`-c`) | Sets the coin type associated with the wallet  | 4219 (=Shimmer) | ✘ | 4218(=IOTA) |
```
Comment (thibault-martinez) [inline documentation/cli/docs/02_wallet.md]: same

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: sdk/examples/client/01_generate_addresses.rs (hunk 6 above)
Comment (Alex6323) [reply sdk/examples/client/01_generate_addresses.rs]: you're right

File: sdk/examples/client/node_api_indexer/00_get_basic_outputs.rs (hunk 7 above)
Comment (Alex6323) [reply sdk/examples/client/node_api_indexer/00_get_basic_outputs.rs]: will change back

**Review by Thoralf-M — DISMISSED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Fmt

**Review by thibault-martinez — APPROVED** [review]

File: cli/src/command/wallet.rs [hunk 10]
```diff
@@ -12,14 +12,27 @@ use log::LevelFilter;
...
+    #[arg(long, value_name = "PATH", env = "WALLET_DATABASE_PATH", default_value = DEFAULT_WALLET_DATABASE_PATH)]
+    pub wallet_db_path: String,
+    /// Set the path to the stronghold snapshot file.
+    #[arg(long, value_name = "PATH", env = "STRONGHOLD_SNAPSHOT_PATH", default_value = DEFAULT_STRONGHOLD_SNAPSHOT_PATH)]
+    pub stronghold_snapshot_path: String,
+    /// Set the account to enter.
+    pub account: Option<String>,
+    /// Set the log level.
+    #[arg(short, long, default_value = DEFAULT_LOG_LEVEL)]
+    pub log_level: LevelFilter,
```
Comment (DaughterOfMars) [inline cli/src/command/wallet.rs:33]: Should we use `RUST_LOG`?

Comment (Alex6323) [reply cli/src/command/wallet.rs:33]: I was also thinking about that, but maybe that's too tough for regular users, dunno.

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**